### PR TITLE
Bug 1310386 - Performance signatures should always be unique

### PR DESCRIPTION
**Pull request by rwood-moz**
This change is [Reviewable](https://reviewable.io/reviews/mozilla/treeherder/2039)

### CODE REVIEW

File: treeherder/etl/perf.py [hunk 1]
```diff
@@ -61,10 +62,14 @@ def _load_perf_datum(job, perf_datum):
         return
     for suite in perf_datum['suites']:
         suite_extra_properties = copy.copy(extra_properties)
+        suite_extra_options = copy.copy(extra_options)
         if suite.get('extraOptions'):
             suite_extra_properties = {
                 'test_options': sorted(suite['extraOptions'])
             }
+            # store extraOptions list as comma-separated str
+            suite_extra_options = ','.join(sorted(suite['extraOptions']))
```
Comment (wlach) [inline treeherder/etl/perf.py]: Let's use a space as a seperator, rather than a comma. It'll look better in the UI.

**Comment by wlach** [comment]
Actually now that I think about it, we should only change the web api's after this initial commit lands (since otherwise there will be an interval where we'd be returning incorrect blank values for extra_options).

That said, I think it would be good to see what that would look like before we go ahead with this. Do you think you could add another commit to this PR with the webapi changes? We can then land this piece-by-piece on stage and then production.

**Comment by rwood-moz** [comment]
Thanks for the review, ok will do!

File: treeherder/webapp/api/performance_data.py [hunk 2]
```diff
@@ -122,6 +122,9 @@ def list(self, request, project):
 
             ret[signature_hash].update(json.loads(extra_properties))
 
+            if extra_options:
+                ret[signature_hash]['extra_options'] = extra_options
```
Comment (wlach) [inline treeherder/webapp/api/performance_data.py]: Just realized this should actually be something like:

```py
ret[signature_hash]['extra_options'] = extra_options.split(' ')
```

Comment (rwood-moz) [inline treeherder/webapp/api/performance_data.py]: Ok so just to confirm. We are storing the new extra_properties in the db as a charField (i.e. 'e10s shell'), however you want the api to not return a char, but to return a list (i.e. ['e10s', 'shell']) instead?

Comment (wlach) [inline treeherder/webapp/api/performance_data.py]: Yup, exactly!

Comment (rwood-moz) [inline treeherder/webapp/api/performance_data.py]: Ok right makes sense, similar to what we do for extra_properties now. Thanks!

**Comment by rwood-moz** [comment]
PR merged manually

**Pull request closed without merging**